### ISSUE TITLE
"Incognito Mode" and anonymous-only profiles (via tor)

### ISSUE THREAD

**Issue by saneki** [qTox/qTox#1350]
A lot of people "criticize" Tox for not providing anonymity by default, although Tox supports communicating through Tor via a proxy. However, installing and starting / setting up Tor may be confusing for people who aren't great with computers even if they find qTox easy to install and use as a graphical program that you install, open and it just works.

It might be worth looking into (optionally?) packaging Tor with qTox on certain platforms, specifically Windows and maybe OSX (I'm unsure as I don't have much experience with OSX). Then at runtime qTox may check if Tor is installed and attempt to setup a SOCKS proxy with it, using that if the user decides to enter "Incognito Mode" or if they use an anonymous-only profile. Of course, qTox would have to handle instances of Tor not being found or the SOCKS proxy server not starting correctly, and also inform the user of the main consequence of anonymity: slower communication.

This is just an idea and how I've described it might not be the best way to go about implementing such a feature. I'm planning to look into it further, but I'm wondering if anyone has any feedback or thinks it isn't such a great idea.

**Comment by zetok** [comment]
What you propose would imply supporting Tox over Tor, which is not happening. Using Tox over Tor is due to Tox supporting proxies.

Edit: please close issue if that's all.

**Comment by saneki** [comment]
I'm not talking about Tox over Tor, just qTox being able to start a Tor proxy and reconnect via the proxy at the click of a button. Some users might not know what Tor is or how to install or use it, yet might still want to be anonymous at times (especially if primarily using qTox for text chat). This would make Tor's provided anonymity more accessible to them.

**Comment by suhr** [comment]
Good idea.

@zetok qBittorrent have anonymous mode though bittorent itself doesn't support tor/i2p.

**Comment by Zer0-One** [comment]
@saneki people that don't know enough about Tor to be able to connect Tox to it themselves are probably better served _without_ the feature you suggest; it might give them a false sense of security while they defeat their own anonymity.

**Comment by saneki** [comment]
@Zer0-One I agree, though anonymous-only profiles might somewhat solve this (profiles that only allow connecting via a Tor proxy server). Also, the user might be prompted with a dialog on the first run of qTox, asking if they want to run qTox anonymously while informing them of the speed penalty and how just one non-anonymous connection may expose their IP address.

**Comment by zetok** [comment]
@suhr 

> qBittorrent have anonymous mode though bittorent itself doesn't support tor/i2p.

From what I remember, that only makes qBittorrent not use same client id, turns off local peer discovery, and some other minor stuff like that. As such it only introduces false sense of anonymity…

@saneki 

> Also, the user might be prompted with a dialog on the first run of qTox, asking if they want to run qTox anonymously while informing them of the speed penalty and how just one non-anonymous connection may expose their IP address.

Overcomplicating things doesn't help user security or anonymity. That kind of option will mislead most of users and will make it harder to use software, without actually providing any benefit beside what is already possible for users who know how to go around things.

And yes, knowledge how to go around things is a **requirement** when it comes to anonymity. Anything less means false sense of gaining anonymity.

Thus what you're proposing doesn't really give any benefit, and introduces only serious drawbacks.

**Comment by suhr** [comment]
> As such it only introduces false sense of anonymity…

It actually implies that user will set a tor/i2p proxy.

**Comment by ghost** [comment]
No I think this wouldn't be useful. Users who really want anonymity can just download the Tor Browser Bundle, run it, configure qTox a bit and that's it. I think this is really not difficult.
And if the user really don't know what Tor is then they can "learn" it. This doesn't hurt and maybe they will also discover other features of Tor. (If I want anonymous chat I supposedly also like surfing via Tor)

So if they really don't know let them discover Tor. Tor is easy to set up, the only thing which is a bit difficult is to set the proxy settings, but with instructions it's also quite easy I think.

However what you can do is to "advertise" this anonymity feature of qTox and improve the instructions e.g. that's a thing which would be useful to make more users aware of ToT (I like this abbrevation :smile:)

**Comment by kradan** [comment]
> Users who really want anonymity can just download the Tor Browser Bundle, run it, configure qTox a bit and that's it. I think this is really not difficult.

This is not true because it is not possible to create an account over Tor (#2070) and there is the UDP in cleartext thing (#3779).

**Comment by anthonybilinski** [comment]
#3779 is now fixed.
